tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -1466,157 +1466,215 @@ def test_copy_object_during_pod_restart(self):
         event.clear()
-        thread.join()
-
-        LOGGER.info("Step 8: Verify status for In-flight READs/WRITEs/DELETEs while pod was"
-                    "restarting are successful without any failures.")
-        responses = {}
-        while len(responses) != 2:
-            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
-        pass_logs = list(x[1] for x in responses["pass_res"])
-        fail_logs = list(x[1] for x in responses["fail_res"])
+        LOGGER.info("Step 8: Verify status for In-flight READs/WRITEs/DELETEs while data pod %s "
+                    "was restarted.", pod_name)
+        LOGGER.info("Waiting for background IOs thread to join")
+        thread_wri.join()
+        thread_rd.join()
+        thread_del.join()
+        LOGGER.info("Step 8.1: Verify status for In-flight DELETEs")
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do deletes")
+        fail_del_bkt = del_resp[1]
+        rem_bkts_aftr_del = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket deletion failed when cluster was online {fail_del_bkt}")
+        assert_utils.assert_true(len(rem_bkts_aftr_del) < del_bucket,
+                                 "Some bucket deletion expected during pods going down")
+        LOGGER.info("Step 8.1: Verified status for In-flight DELETEs")
+        LOGGER.info("Step 8.2: Verify status for In-flight WRITEs")
+        responses_wr = dict()
+        while len(responses_wr) != 2:
+            responses_wr = output_wr.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses_wr["pass_res"])
+        fail_logs = list(x[1] for x in responses_wr["fail_res"])
         resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
-        assert_utils.assert_false(len(resp[1]),
-                                  f"IOs during no pod restart contain failures: {resp[1]}")
-        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs)
-        assert_utils.assert_false(len(resp[1]),
-                                  f"IOs during pod restart contain failures: {resp[1]}")
-        LOGGER.info("Step 8: Verified status for In-flight READs/WRITEs/DELETEs while pod was"
-                    "restarting are successful without any failures.")
-
+        assert_utils.assert_false(len(resp[1]), f"Logs which contain failures: {resp[1]}")
+        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs, pass_logs=False)
+        # TODO: Expecting Failures when data pods going down. Re-test once CORTX-28541 is Resolved
+        assert_utils.assert_true(len(resp[1]) <= len(fail_logs),
+                                 f"Logs which contain pass: {resp[1]}")
```

Comment:
Isn't this data pod restart? We expect all IOs to be passed right?

Reply:
yeah my bad, changed at all places